### ISSUE TITLE
Masking images in Exercise 1.2.2

### ISSUE THREAD

**Issue by jonathanloganmoran** [jonathanloganmoran/ND0013-Self-Driving-Car-Engineer#3]
In Sect. 2.3 of [Exercise 1.2.2](https://github.com/jonathanloganmoran/ND0013-Self-Driving-Car-Engineer/blob/4214eaf2bfd1d383aca0985458ac2f97efd9acf9/1-Object-Detection-in-Urban-Environments/Exercises/1-2-2-Image-Manipulation/2022-08-17-Image-Manipulation-Masking.ipynb) I am running into the following issue with `mask_and_display()` while attempting to mask the colour images provided to me by Udacity.

![Fig. 1. Issue displaying masked RGB image in Matplotlib for Udacity test image.](https://user-images.githubusercontent.com/22329957/186269783-ce443b30-adca-49e4-8953-0044f942be87.png)

Shown in the above output on the right subplot is the _masked image_ output currently computed with
```
img_masked = cv2.bitwise_and(img, img, mask)
```

The `img_masked` output has shape `(1280, 1920, 3)` which matches the original `img`. 

When converting to BGR using
```
cvt.Color(img_masked, cv2.COLOR_RGB2BGR) 
```
Matplotlib displays the masked image using the appropriate colormap (not the orange-tinted shown). While the result matches the colouring of the original image (left), it does not exhibit the expected masked image characteristics i.e., `(0,0,0)` where `mask == 0`. 

Trying the alternative method for masking images, as suggested by Udacity,
```
img_masked_out = img * np.stack([mask]*3, axis=2)
```
does not result in any difference to what is plotted above.

**Comment by jonathanloganmoran** [comment]
#### UPDATE 2022-08-23 14:45 PST:
While extremely odd, it is worth noting that I am _not_ running into the same issue when plotting the masked image generated from my own sample image, shown below:

![Fig. 2. No issues displaying masked RGB image in Matplotlib for own test image.](https://user-images.githubusercontent.com/22329957/186272195-c9b8ceaf-5c17-4ae8-900e-9b40f1eef310.png)

**Comment by jonathanloganmoran** [comment]
#### UPDATE 2022-10-19 14:31 PT:

Closing as `wontfix` for now; bug is isolated to the specific image used.

**Comment by jonathanloganmoran** [comment]
#### UPDATE 2023-08-15 13:49 PT:

Re-opening this issue since it has been addressed by @heng-yuwen in PR #49.
